uvicorn/main.py

```diff
@@ -33,8 +34,39 @@
 )
 
 
+def _setup_defaults_from_settings_script(
+    ctx: click.Context, param: click.Parameter, value: typing.Optional[str]
+):
+    if value is None:
+        return
+
+    members = runpy.run_path(value)  # type: dict
+
+    try:
+        config = members["config"]  # type: Config
+    except KeyError:
+        raise click.BadParameter("no 'config' object declared in {}".format(value))
+
+    if not isinstance(config, Config):
+        raise click.BadParameter(
+            "'config' must be a 'Config' object, not '{}'".format(type(config).__name__)
+        )
+
+    # Read parameter defaults from the config.
+    # Values can still be overridden using command line arguments.
+    ctx.default_map = vars(config)
```

Comment:
I'm not convinced that `vars(config)` here would necessarily map exactly onto the `main()` arguments.

Reply:
Yes, I can see why. I provided a fix that explicitly reads the dict items from the command's parameters. :+1: